desktop-environment.el

```diff
@@ -182,6 +192,49 @@ portion of the screen."
   (message "New volume value: %s" (desktop-environment-volume-get)))
 
 
+;;; Helper functions - keyboard backlight
+(defun desktop-environment-keyboard-backlight-percent ()
+  "Return the new keyboard backlight value as a % of maximum backlight."
+  (let ((backlight-level (desktop-environment-keyboard-backlight-get)))
+    (if (eq backlight-level 0)
+        (progn (let ((percent "0.0"))
+                 (save-match-data percent)))
+      (progn (let ((percent (*
+                             (/ (* backlight-level 1.0)
+                                (* (desktop-environment-keyboard-backlight-get-max) 1.0))
+                             100)))
+               (save-match-data percent))))))
+
+(defun desktop-environment-keyboard-backlight-get ()
+  "Return a string representing keyboard backlight current level."
+  (let ((output (dbus-call-method :system
+                                   "org.freedesktop.UPower"
+                                   "/org/freedesktop/UPower/KbdBacklight"
+                                   "org.freedesktop.UPower.KbdBacklight"
+                                   "GetBrightness")))
+    (save-match-data output)))
+
+(defun desktop-environment-keyboard-backlight-get-max ()
+  "Return a string representing keyboard backlight maximum level."
+  (let ((output (dbus-call-method :system
+                                   "org.freedesktop.UPower"
+                                   "/org/freedesktop/UPower/KbdBacklight"
+                                   "org.freedesktop.UPower.KbdBacklight"
+                                   "GetMaxBrightness")))
+    (save-match-data output)))
+
+(defun desktop-environment-keyboard-backlight-set (value)
+  "Set keyboard backlight to VALUE."
+  (let ((output (+ (desktop-environment-keyboard-backlight-get) (string-to-number value))))
```

Comment:
I would give the responsibility of the addition to the callers of this function for maximum reusability. That way, the behavior of the function would be closer to what the function name and docstring actually imply. For example, I would write something like that instead:

```emacs
(defun desktop-environment-keyboard-backlight-set (value)
  "Set keyboard backlight to VALUE."
  (dbus-call-method :system
                      "org.freedesktop.UPower"
                      "/org/freedesktop/UPower/KbdBacklight"
                      "org.freedesktop.UPower.KbdBacklight"
                      "SetBrightness"
                      :int32 value))
  (message "New keyboard value: %s%%" (desktop-environment-keyboard-backlight-percent)))
```

and then

```emacs
(defun desktop-environment-keyboard-backlight-increment ()
  "Increment keyboard backlight by `desktop-environment-keyboard-backlight-normal-increment`."
  (interactive)
  (desktop-environment-keyboard-backlight-set
    (+ desktop-environment-keyboard-backlight-normal-increment
        (desktop-environment-keyboard-backlight-get)))
```

I know I'm very picky so you can ignore this feedback :-).

Reply:
Quite the opposite, I'm very glad to make use of your feedback. It's well-thought-out.